table/updates.go

```diff
@@ -452,7 +458,7 @@ type removeSnapshotRefUpdate struct {
 
 // NewRemoveSnapshotRefUpdate creates a new update that removes a snapshot reference
 // from the table metadata.
-func NewRemoveSnapshotRefUpdate(ref string) *removeSnapshotRefUpdate {
+func NewRemoveSnapshotRefUpdate(ref string) Update {
```

Comment:
Go convention, where possible, is to return the concrete type, not the interface